src/model/transactions.coffee

```diff
@@ -38,17 +38,19 @@ RouteMetadataDef =
 
 # Trasnaction schema
 TransactionSchema = new Schema
-  "clientID":       Schema.Types.ObjectId
-  "clientIP":       String
-  "parentID":       Schema.Types.ObjectId
-  "childIDs":       [Schema.Types.ObjectId]
-  "channelID":      type: Schema.Types.ObjectId, index: true
-  "request":        RequestDef
-  "response":       ResponseDef
-  "routes":         [RouteMetadataDef]
-  "orchestrations": [OrchestrationMetadataDef]
-  "properties":     Object
-  "canRerun":       type: Boolean, default: true
+  "clientID":             Schema.Types.ObjectId
+  "clientIP":             String
+  "parentID":             Schema.Types.ObjectId
+  "childIDs":             [Schema.Types.ObjectId]
+  "channelID":            type: Schema.Types.ObjectId, index: true
+  "request":              RequestDef
+  "response":             ResponseDef
+  "routes":               [RouteMetadataDef]
+  "orchestrations":       [OrchestrationMetadataDef]
+  "properties":           Object
+  "canRerun":             type: Boolean, default: true
+  "wasRerun":             type: Boolean, default: false
+  "internalServerError":  Boolean
```

Comment:
I think for the real implementation we should handle internal errors better. Both in the core and console. I think storing the error text would be very useful as well as handling of mediator returning internal errors.